Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -0,0 +1,303 @@
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// TextBox Mask property allows a user to more easily enter fixed width text in TextBox control
+    /// where you would like them to enter the data in a certain format
+    /// </summary>
+    public partial class TextBoxMask
+    {
+        private const string DefaultPlaceHolder = "_";
+        private static readonly KeyValuePair<char, string> AlphaCharacterRepresentation = new KeyValuePair<char, string>('a', "[A-Za-z]");
+        private static readonly KeyValuePair<char, string> NumericCharacterRepresentation = new KeyValuePair<char, string>('9', "[0-9]");
+        private static readonly KeyValuePair<char, string> AlphaNumericRepresentation = new KeyValuePair<char, string>('*', "[A-Za-z0-9]");
+
+        private static void OnMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnPlaceHolderChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnCustomMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void InitTextBoxMask(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var textbox = d as TextBox;
+
+            // incase no value is provided us it as normal textbox
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            if (string.IsNullOrWhiteSpace(mask))
+            {
+                return;
+            }
+
+            var placeHolderValue = textbox.GetValue(PlaceHolderProperty) as string;
+            if (string.IsNullOrEmpty(placeHolderValue))
+            {
+                return;
+            }
+
+            var placeHolder = placeHolderValue[0];
+
+            var representationDictionary = textbox.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            representationDictionary = new Dictionary<char, string>();
+            representationDictionary.Add(AlphaCharacterRepresentation.Key, AlphaCharacterRepresentation.Value);
+            representationDictionary.Add(NumericCharacterRepresentation.Key, NumericCharacterRepresentation.Value);
+            representationDictionary.Add(AlphaNumericRepresentation.Key, AlphaNumericRepresentation.Value);
+
+            var customDictionaryValue = textbox.GetValue(CustomMaskProperty) as string;
+            if (!string.IsNullOrWhiteSpace(customDictionaryValue))
+            {
+                var customRoles = customDictionaryValue.Split(',');
+                foreach (var role in customRoles)
+                {
+                    var roleValues = role.Split(':');
+                    if (roleValues.Length != 2)
+                    {
+                        throw new Exception("Invalid custom mask");
```

Comment:
Should we throw an InvalidOperationException (or something similar) here instead of a basic Exception?